packages/table-core/src/core.tsx

```diff
@@ -331,7 +332,7 @@ export function createTableInstance<TGenerics extends AnyGenerics>(
         accessorFn = columnDef.accessorFn
       } else if (columnDef.accessorKey) {
         accessorFn = (originalRow?: TGenerics['Row']) =>
-          (originalRow as any)[columnDef.accessorKey]
+          getProp(originalRow as any, columnDef.accessorKey)
```

Comment:
get the nested property, rather than expecting data like `{ 'accessor.accessor': value }`